language/move-lang/src/expansion/dependency_ordering.rs

```diff
@@ -38,6 +44,35 @@ pub fn verify(
             }
         }
     }
+
+    for (node, neighbor_nodes) in module_neighbors {
+        for (neighbor_node, relations) in neighbor_nodes {
+            for dep_type in relations.keys() {
+                match dep_type {
+                    DepType::Use => modules
+                        .get_mut(&node)
+                        .unwrap()
+                        .dependency_summary
+                        .insert(neighbor_node.clone()),
+                    DepType::Friend => modules
+                        .get_mut(&neighbor_node)
+                        .unwrap()
+                        .dependency_summary
+                        .insert(node.clone()),
+                };
+            }
+        }
+    }
```

Comment:
This feels non optimal. I would just record this information during `add_neighbor` and extract it when you are done